app/socket-apis/__tests__/subscribe-deliberation.js

```diff
@@ -0,0 +1,149 @@
+  getStatementIds,
+  initDiscussion,
+  insertStatementId,
+  putGroupings,
+  rankMostImportant,
+} from '../../dturn/dturn'
+import upsertPoint from '../upsert-point'
+
+const handle = 'subscribe-deliberation'
+const socketApiUnderTest = subscribeDeliberation
+
+const userId = '6667d5a33da5d19ddc304a6b'
+const synuser = { synuser: { id: userId } }
+
+const discussionId = '66a174b0c3f2051ad387d2a6'
+
+let MemoryServer
+
+import subscribeDeliberation from '../subscribe-deliberation'
+
+beforeEach(async () => {
+  jest.spyOn(console, 'error').mockImplementation(() => {})
+})
+
+afterEach(async () => {
+  endTest = false
+  console.error.mockRestore()
+})
+
+beforeAll(async () => {
+  MemoryServer = await MongoMemoryServer.create()
+  const uri = MemoryServer.getUri()
+  await Mongo.connect(uri)
+
+  await jestSocketApiSetup(userId, [[handle, socketApiUnderTest]])
+})
+
+afterAll(async () => {
+  Mongo.disconnect()
+  MemoryServer.stop()
+  window.socket.close()
+})
+
+// Tests
+test('Fail if user is not logged in.', async () => {
+  await subscribeDeliberation.call({}, discussionId)
+
+  expect(console.error.mock.calls[0][0]).toMatch(/user is not logged in/)
+})
+
+test('Fail if deliberation ID not provided.', async () => {
+  await subscribeDeliberation.call(synuser, null)
+
+  expect(console.error.mock.calls[0][0]).toMatch(/DeliberationId was not provided/)
+})
+// this needs to be outside of any tests because it needs to be passed in through one test, and will get executed later when other tests trigger it
+let updateHandlerDone
+let endTest = false
+function updateHandler(data) {
+  if (endTest === true) {
+    if (updateHandlerDone) updateHandlerDone(data)
+    else console.error('updatehandler called, but updateHandlerDone was not set')
+  }
+}
+
+test('Successful request if deliberation exists.', done => {
+  const anIota = {
+    _id: new ObjectId(discussionId),
+    path: '/deliberation1',
+    subject: 'Deliberation',
+    description: 'A descriptive discussion.',
+    webComponent: {
+      dturn: {
+        max_rounds: 3,
+      },
+    },
+  }
+
+  Iota.create(anIota)
+    .then(() => {
+      function requestHandler(data) {
+        done()
+      }
+
+      socketApiSubscribe(handle, discussionId, requestHandler, updateHandler)
+    })
+    .catch(err => done(err))
+})
+
+test('Check updateHandler is called.', done => {
+  endTest = true
+
+  updateHandlerDone = data => {
+    expect(data).toEqual({ participants: 1, lastRound: 0 })
+    done()
+  }
+  const pointId = new ObjectId()
+  const pointObj = { _id: pointId, title: 'Point 1', description: 'Description 1' }
+
+  // this will trigger the update handler above
+  upsertPoint
+    .call(synuser, pointObj, () => {})
+    .then(() => {
+      const insertResult = insertStatementId(discussionId, userId, pointId)
+      expect(insertResult).toBe(pointId)
+    })
+})
+
+test('Check lastRound update.', async done => {
+  let num
+
+  updateHandlerDone = async data => {
```

Comment:
Instead of endTest, updateHandlerDone cold be set to ()=>{} here, and down below, when endTest is set to ture, updateHandlerDone could be set to what's here.  

But this all looks cleaner than the counter was before.